cmd/retester/main_test.go

```
@@ -1,19 +1,125 @@
+			expected: options{
+				runOnce:        false,
+				dryRun:         true,
+				interval:       hour,
+				cacheFile:      "",
+				cacheRecordAge: sevenDays,
+				configFile:     "",
+			},
+			expectFatal: false,
+		},
+		{
+			name: "basic case",
+			args: []string{"cmd", "--run-once=true", "--interval=2h", "--cache-file=cache.yaml", "--cache-record-age=100h", "--config-file=config.yaml"},
+			expected: options{
+				runOnce:        true,
+				dryRun:         true,
+				interval:       2 * hour,
+				cacheFile:      "cache.yaml",
+				cacheRecordAge: 100 * hour,
+				configFile:     "config.yaml",
+			},
+			expectFatal: false,
+		},
+		{
+			name:        "wrong interval and empty cache record age",
+			args:        []string{"cmd", "--interval=notNumber", "--cache-record-age="},
+			expected:    options{dryRun: true},
+			expectFatal: true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			fatalErr = false
+			os.Args = tc.args
+			actual := gatherOptions()
+			if diff := cmp.Diff(tc.expectFatal, fatalErr); diff != "" {
+				t.Errorf("Unexpected fatal error:\n%s", diff)
+			} else {
+				opts := []cmp.Option{
+					cmpopts.IgnoreFields(options{}, "config", "github"),
+					cmp.AllowUnexported(options{}),
+				}
+				if diff := cmp.Diff(tc.expected, actual, opts...); diff != "" {
+					t.Errorf("%s differs from expected:\n%s", tc.name, diff)
+				}
+			}
+		})
+	}
+}
 
-	if diff := cmp.Diff(expected, actual.configFile); diff != "" {
-		t.Errorf("Test failed, expected: '%+v', got:  '%+v'", expected, actual.configFile)
+func TestValidate(t *testing.T) {
+	testCases := []struct {
+		name     string
+		o        options
+		expected error
+	}{
+		{
+			name: "basic",
+			o: options{
+				config:         flagutil.ConfigOptions{ConfigPath: "/etc/config/config.yaml"},
+				runOnce:        false,
+				dryRun:         true,
+				interval:       hour,
+				cacheFile:      "",
+				cacheRecordAge: sevenDays,
+				configFile:     "",
```

Should we raise an error for this case?

---

I think we need configFile right?